packages/driver/src/cy/testConfigOverrides.ts

```diff
@@ -46,24 +107,38 @@ function mutateConfiguration (testConfigOverride, config, env) {
 
 // this is called during test onRunnable time
 // in order to resolve the test config upfront before test runs
-export function getResolvedTestConfigOverride (test) {
+// note: must return as an object to meet the dashboard recording API
+export function getResolvedTestConfigOverride (test): ResolvedTestConfigOverride {
   let curParent = test.parent
 
-  const testConfig = [test._testConfig]
+  const testConfigList = [{
+    overrides: test._testConfig,
+    invocationDetails: test.invocationDetails,
+  }]
 
   while (curParent) {
     if (curParent._testConfig) {
-      testConfig.push(curParent._testConfig)
+      testConfigList.unshift({
+        overrides: curParent._testConfig,
+        invocationDetails: curParent.invocationDetails,
+      })
     }
 
     curParent = curParent.parent
   }
```

Comment:
If this is what's causing the `driver` tests to hang, it's probably worth rewriting this as recursive instead of a loop. We try to opt for recursion over looping wherever possible because eventually it will hit max stack depth and error, whereas an infinite loop just causes this freezing behavior that's harder to debug. Rewriting it as recursive means if there's another bug at least we get a stack trace.

Reply:
This wasn't what was causing the tests to hang. What was causing our tests to hang was the fact the driver tests has a global `beforeEach` hook and a test that set baseUrl to an empty string (ironic given the issue that started this work 😄 ). What was happening was the configuration validation was executing and throwing the error outside of the test context. This caused `driver/src/cypress/cy` to capture the error as an onTop error. When this error is thrown, the runner essential hangs & the test doesn't finish. CircleCi was eventually timing out.